opencensus/stats/aggregation_data.py

```diff
@@ -123,34 +162,37 @@ def __init__(self,
                  counts_per_bucket=None,
                  bounds=None,
                  exemplars=None):
+        if bounds is None and exemplars is not None:
+            raise ValueError
+        if exemplars is not None and len(exemplars) != len(bounds) + 1:
+            raise ValueError
+
         super(DistributionAggregationData, self).__init__(mean_data)
         self._mean_data = mean_data
         self._count_data = count_data
         self._min = min_
         self._max = max_
         self._sum_of_sqd_deviations = sum_of_sqd_deviations
+
         if bounds is None:
             bounds = []
+            self._exemplars = None
         else:
             assert bounds == list(sorted(set(bounds)))
             assert all(bb > 0 for bb in bounds)
+            if exemplars is None:
+                self._exemplars = {ii: None for ii in range(len(bounds) + 1)}
+            else:
+                self._exemplars = {ii: ex for ii, ex in enumerate(exemplars)}
+        self._bounds = (bucket_boundaries.BucketBoundaries(boundaries=bounds)
+                        .boundaries)
 
         if counts_per_bucket is None:
             counts_per_bucket = [0 for ii in range(len(bounds) + 1)]
         else:
             assert all(cc >= 0 for cc in counts_per_bucket)
             assert len(counts_per_bucket) == len(bounds) + 1
-
         self._counts_per_bucket = counts_per_bucket
-        self._bounds = bucket_boundaries.BucketBoundaries(
-            boundaries=bounds).boundaries
-        bucket = 0
-        for _ in self.bounds:
-            bucket = bucket + 1
-
-        # If there is no histogram, do not record an exemplar
-        self._exemplars = \
-            {bucket: exemplars} if len(self._bounds) > 0 else None
```

Comment:
I'm assuming this is a typo and this line should have been indented. Score one for semantic whitespace detractors.